prime_server/prime_server.hpp

```diff
@@ -75,8 +79,7 @@ class server_t {
            bool log = false,
            size_t max_request_size = DEFAULT_MAX_REQUEST_SIZE,
            uint32_t request_timeout = DEFAULT_REQUEST_TIMEOUT,
-           const health_check_matcher_t& health_check_matcher = {},
-           const std::string& health_check_response = {});
+           const shortcircuiter_t<request_container_t>& shortcircuiters = nullptr);
```

Comment:
its not a pointer is an std::function, and we want a default initialized one which has pointer-like semantics (it feels like a function pointer to us)
```suggestion
           const shortcircuiter_t<request_container_t>& shortcircuiters = {});
```